### DISCUSSION TITLE
Does i18next work with vite react?

### DISCUSSION THREAD

**Discussion by RhysOwennn** [i18next/i18next#1807] [Q&A]
I'm getting an error. See my question on stackoverflow. Any help would be much appreciated. https://stackoverflow.com/questions/73080329/how-to-use-i18n-library-with-react-vite

**Comment by jamuhl** [comment]
do you get any other console output? current guess...

a) you forgot to import i18n.js
b) the path to load translations is set wrong (there is no folder locales in public)

**Comment by RhysOwennn** [comment]
Here's a log:
```
Failed to load resource: the server responded with a status of 404 (Not Found)
i18next.js:27 i18next::backendConnector: loading namespace translation for language en failed failed loading /locales/en/translation.json; status code: 404
output @ i18next.js:27
i18next.js:27 i18next: languageChanged en
i18next.js:27 i18next: initialized Object
2i18next.js:27 i18next::translator: missingKey en translation suche suche
2VM126:236 i18next::translator: missingKey en translation suche suche
2i18next.js:27 i18next::translator: missingKey en translation suche suche
2VM126:236 
```
Folder structure looks as follows
![folder structure](https://user-images.githubusercontent.com/69479023/180465853-b2356e1a-bcaa-4865-ade3-1a1cbb5f0e5f.png)

I import it here 
![import](https://user-images.githubusercontent.com/69479023/180466203-6624388d-f169-4514-b918-82d097648c7e.png)

Though I see it says:
`Could not find a declaration file for module '../i18n'. i18n.js' implicitly has an 'any' type.`

**Reply by RhysOwennn** [reply]
I changed it to import ".`./i18n.js";
` 
fixed the module declaration error but still getting the key error in the console.

**Reply by RhysOwennn** [reply]
When I click the button I get 
`i18next: languageChanged de`
so the module has loaded but for some reason the keys aren't found.

**Reply by jamuhl** [reply]
there is one problem in your console log -> the one with backendConnector...must be related to not finding your translation files...

looking at https://vitejs.dev/guide/assets.html#the-public-directory public folder should be accessible...can you open the JSON files in the browser?

**Reply by RhysOwennn** [reply]
You mean like `http://localhost:3000/open-contract/locales/de/common.json` ? That gives a 404error.

**Reply by jamuhl** [reply]
how to you get to that url...http://localhost:3000/locales/de/translation.json <- the file in your public folder

**Reply by RhysOwennn** [reply]
The url with the translation is 
`http://localhost:3000/open-contract/`
So I just added in the browser url `/locales/de/common.json`

**Reply by jamuhl** [reply]
why common.json?  

it says: i18next.js:27 i18next::backendConnector: loading namespace translation for language en failed failed loading /locales/en/translation.json

/locales/en/translation.json -> is looked from the root of your site...

**Reply by RhysOwennn** [reply]
I have both the files common.json, and translation.json
![file structure](https://user-images.githubusercontent.com/69479023/180723345-51095019-8d6c-41ae-90ec-bdb09c484174.png)

**Reply by jamuhl** [reply]
In the end - makes no difference...the files can't be reached from the browser...so this is no issue of i18next but your code...only you can solve that:

`i18next.js:27 i18next::backendConnector: loading namespace translation for language en failed failed loading /locales/en/translation.json`

is explicit error telling you what is wrong...and you should be able to solve this.

**Reply by RhysOwennn** [reply]
So you're saying the issue is with my pathing? Okay, I will look into it.

**Reply by jamuhl** [reply]
I saying that since the start of discussion...either is the path wrong or your dev server does not statically serve that folder inside public folder

**Comment by iannak** [comment]
Does i18n have to be at the root of the project? Does it work with TS or just JS?
<img width="822" alt="Screenshot 2023-09-28 at 11 15 32" src="https://github.com/i18next/i18next/assets/63086797/6481810f-92fe-4d76-990c-27b2525c935a">

**Reply by addayounes** [reply]
What is your vscode theme ?

**Reply by addayounes** [reply]
What is your vscode theme?

**Reply by fiqrisr** [reply]
what icon theme is that? @iannak

**Reply by iannak** [reply]
> what icon theme is that? @https://marketplace.visualstudio.com/items?itemName=johnpapa.winteriscoming

**Reply by iannak** [reply]
> What is your vscode https://marketplace.visualstudio.com/items?itemName=johnpapa.winteriscoming
